sns/src/main/scala/akka/stream/alpakka/sns/javadsl/SnsPublisher.scala

```diff
@@ -33,7 +33,13 @@ object SnsPublisher {
    * creates a [[akka.stream.javadsl.Flow Flow]] to publish messages to a SNS topic using an [[com.amazonaws.services.sns.AmazonSNSAsync AmazonSNSAsync]]
    */
   def createPublishFlow(topicArn: String, snsClient: AmazonSNSAsync): Flow[PublishRequest, PublishResult, NotUsed] =
-    Flow.fromGraph(new SnsPublishFlowStage(topicArn, snsClient))
+    Flow.fromGraph(new SnsPublishFlowStage(Some(topicArn), snsClient))
+
+  /**
+   * creates a [[akka.stream.javadsl.Flow Flow]] to publish messages to a SNS topic using an [[com.amazonaws.services.sns.AmazonSNSAsync AmazonSNSAsync]]
```

Comment:
Update the Scaladoc to reflect the difference between the two factory methods, please.